components/region_cache_memory_engine/src/write_batch.rs

```diff
@@ -137,6 +143,32 @@ impl RangeCacheWriteBatchEntry {
         range_manager.contains(&self.key)
     }
 
+    #[inline]
+    pub fn maybe_cached(&self, seq: u64, engine_core: &mut RangeCacheMemoryEngineCore) {
+        for r in &engine_core.range_manager().pending_ranges_with_snapshot {
```

Comment:
I think this should be refined with log(n) algorithm, instead of O(n) check. 
What's more, for every write and very key to go through this, it's not inefficient. I guess we may have to do it in region level (that is for each region's round in apply thread, this check is done only once)

Reply:
We will limit the total amount of ranges to load at the same time, say 5. So linear scan is fine.

> What's more, for every write and very key to go through this, it's not inefficient.
At least for now, it's not called for every write. It will only be checked at `write_to_db` just like checking whether to write a key in the memory engine.